tools/pipelines/templates/build-npm-client-package.yml

```
@@ -0,0 +1,898 @@
+              outputs:
+                - ${{ if eq(parameters.testCoverage, true) }}:
+                      - output: pipelineArtifact
+                        displayName: Publish Artifacts - code-coverage
+                        condition: and( succeededOrFailed(), eq(variables['ReportDirExists'], 'true'))
+                        targetPath: $(Build.ArtifactStagingDirectory)/codeCoverageAnalysis
+                        artifactName: 'codeCoverageAnalysis-$(System.JobAttempt)'
+                        sbomEnabled: false
+                        publishLocation: pipeline
+
+          # Job - Tests
+          - ${{ each test in parameters.taskTest }}:
+            - job: Test_${{ test.jobName }}
+              displayName: "Run Task Test ${{ test.jobName }}"
+              dependsOn: build
+              variables:
+                - group: ado-feeds
+                - group: storage-vars
+                - name: releaseBuildVar
+                  value: '$[variables.releaseBuild]'
+                - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+                  - name: targetBranchName
+                    value: $(System.PullRequest.TargetBranch)
+                # Absolute path to the folder that contains the source code for the telemetry-generator package, which is
+                # used in a few places in the pipeline to push custom telemetry to Kusto.
+                - name: absolutePathToTelemetryGenerator
+                  value: $(Build.SourcesDirectory)/tools/telemetry-generator
+                  readonly: true
+              steps:
+                # Setup
+                - checkout: self
+                  clean: true
+                  lfs: '${{ parameters.checkoutSubmodules }}'
+                  submodules: '${{ parameters.checkoutSubmodules }}'
+
+                - template: /tools/pipelines/templates/include-use-node-version.yml@self
+
+                - template: /tools/pipelines/templates/include-install.yml@self
+                  parameters:
+                    packageManager: '${{ parameters.packageManager }}'
+                    buildDirectory: '${{ parameters.buildDirectory }}'
+                    packageManagerInstallCommand: '${{ parameters.packageManagerInstallCommand }}'
+
+                - task: DownloadPipelineArtifact@2
+                  inputs:
+                    artifact: nested_lib_dist
+                    targetPath: $(Build.SourcesDirectory)
+
+                - script: |
+                    echo "Extracting nested lib/dist directories and their contents..."
+                    tar --extract --gzip --file $(Build.SourcesDirectory)/nested_lib_dist.tar.gz --directory $(Build.SourcesDirectory)
+                  displayName: Extract Nested Lib/Dist Directories and Their Contents
+
+                # Test
+
+                # Set variable startTest if the build succeed so that we can run all the test tasks whether they are failed or not
```

Also, I think we should move this task and its comment to be below the conditional build in case of jest tests. If we set this variable to true, then that build fails, tests probably shouldn't be running so we shouldn't have set the variable yet.
```suggestion
                 # Set variable startTest if everything is good so far and we'll start running tests,
                 # so that the steps to process/upload test coverage results only run if we got to the point of actually running tests.
```